src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/Semantics/Types/TypeManager.cs

```diff
@@ -1112,12 +1112,11 @@ internal bool GetBestAccessibleType(CSemanticChecker semanticChecker, BindingCon
                 return true;
             }
 
-            Debug.Assert(typeSrc.IsAggregateType());
+            Debug.Assert(typeSrc is AggregateType);
 
```

Comment:
This assert means that we expect the condition to always be true. Why does the conditional below exist? is this a fallout from a previous refactoring?